src/test/java/org/openstreetmap/atlas/utilities/collections/ShardBucketCollectionTest.java

```diff
@@ -0,0 +1,297 @@
+
+        // test getting buckets
+        polylineBuckets2.add(firstBucketPolyline);
+        polylineBuckets2.add(lastBucketPolyline);
+        Assert.assertFalse(polylineBuckets2.isEmpty());
+        Assert.assertEquals(2, polylineBuckets2.size());
+        Assert.assertTrue("Bucket does not exist or doesn't contain polyline",
+                polylineBuckets2.getBucketCollectionForShard(SlippyTile.forName("3-0-0"))
+                        .map(collection -> collection.contains(firstBucketPolyline)).orElse(false));
+        Assert.assertTrue(
+                polylineBuckets2.getBucketCollectionsForBounds(lastBucketPolyline.bounds())
+                        .allMatch(collection -> collection.contains(lastBucketPolyline)));
+
+        // outside of bounds shard
+        Assert.assertFalse(polylineBuckets2.getBucketCollectionForShard(SlippyTile.forName("3-4-0"))
+                .isPresent());
+        // wrong zoom shard
+        Assert.assertFalse(polylineBuckets2.getBucketCollectionForShard(SlippyTile.forName("4-0-0"))
+                .isPresent());
+
+        // test remove
+        Assert.assertTrue(polylineBuckets2
+                .containsAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+        Assert.assertTrue(
+                polylineBuckets2.removeAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+        Assert.assertFalse(polylineBuckets2.remove(firstBucketPolyline));
+        Assert.assertFalse(polylineBuckets2.remove(new Integer(0)));
+        Assert.assertFalse(polylineBuckets2
+                .containsAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+
+        // test retain
+        Assert.assertTrue(
+                polylineBuckets2.addAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+        Assert.assertFalse(
+                polylineBuckets2.retainAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+        Assert.assertTrue(polylineBuckets2
+                .containsAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline)));
+        Assert.assertTrue(polylineBuckets2.retainAll(Arrays.asList(firstBucketPolyline)));
+        Assert.assertTrue(polylineBuckets2.containsAll(Arrays.asList(firstBucketPolyline)));
+
+        // test array transfer
+        polylineBuckets1.clear();
+        polylineBuckets2.clear();
+        polylineBuckets1.add(firstBucketPolyline);
+        polylineBuckets2.addAll(Arrays.asList(firstBucketPolyline, lastBucketPolyline));
+        final Object[] firstCollectionArray = polylineBuckets1.toArray();
+        final PolyLine[] array1 = new PolyLine[] { firstBucketPolyline };
+        Assert.assertArrayEquals(array1, firstCollectionArray);
+        final PolyLine[] array2 = new PolyLine[] { firstBucketPolyline, firstBucketPolyline,
+                lastBucketPolyline };
+        final PolyLine[] allArray = polylineBuckets2.toArray(array1);
+        Assert.assertArrayEquals(array2, allArray);
+
+        // get code coverage
```

Comment:
What does get code coverage mean?

Reply:
I want to use all the methods to get good coverage score for sonar.